### PR TITLE
Add time varying error model function

### PR DESCRIPTION
**Pull request by ZheHuang96**
Hi Rikard, 

Here is the add_time_varying_error function. After this task, I am going to change the models and workflow in resmod:)

### CODE REVIEW

File: src/pharmpy/modeling/error.py [hunk 1]
```diff
@@ -597,3 +597,47 @@ def set_dtbs_error_model(model, fix_to_log=False):
...
+
+
+def add_time_varying_error_model(model, cutoff, idv='TIME'):
+    """Set a time varying error model per time cutoff
+
+    Parameters
+    ----------
+    model : Model
+        Pharmpy model
+    cutoff : Float
```
Comment (rikardn) [inline src/pharmpy/modeling/error.py:609]: `Float` should be `float`

File: src/pharmpy/modeling/error.py [hunk 2]
```diff
@@ -597,3 +597,47 @@ def set_dtbs_error_model(model, fix_to_log=False):
...
+        A value at the given quantile over idv column
+    idv : String
```
Comment (rikardn) [inline src/pharmpy/modeling/error.py:611]: `String` should be `str`  (`str` being the name of the string class)

File: src/pharmpy/modeling/error.py [hunk 3]
```diff
@@ -597,3 +597,47 @@ def set_dtbs_error_model(model, fix_to_log=False):
...
+        Time or time after dose, default is Time
+
+    Return
+    ------
+    Model
```
Comment (rikardn) [inline src/pharmpy/modeling/error.py:616]: Add the explanation line of the Model as in the other functions

File: src/pharmpy/modeling/error.py [hunk 4]
```diff
@@ -597,3 +597,47 @@ def set_dtbs_error_model(model, fix_to_log=False):
     model.observation_transformation = obs
 
     return model
+
+
+def add_time_varying_error_model(model, cutoff, idv='TIME'):
```
Comment (rikardn) [inline src/pharmpy/modeling/error.py:602]: Perhaps we should rename it to `set_time_varying_error_model` as we actually change the model to become time varying.

**Review by rikardn — CHANGES_REQUESTED** [review]
Looks good in general. Some minor fixes needed.

**Comment by rikardn** [comment]
Also check the failing doctest. Seems to be a space too much in the output given in the docstring

**Comment by ZheHuang96** [comment]
No problem, I can fix the problems now. I was checking the failing doctest for py39 and feel confusing about it... Thank you for your comment, I am going to check the space of the output!

**Comment by ZheHuang96** [comment]
I am sorry I press the wrong button

File: src/pharmpy/modeling/error.py [hunk 5]
```diff
@@ -597,3 +597,49 @@ def set_dtbs_error_model(model, fix_to_log=False):
...
+        Reference to the same model object
+
+    Examples
+    --------
+    >>> from pharmpy.modeling import load_example_model, set_time_varying_error_model
+    >>> model = load_example_model("pheno")
+    >>> set_time_varying_error_model(model, cutoff=1.0)    # doctest: +ELLIPSIS
+    <...>
+    >>> model.statements.find_assignment("Y")
+          ⎧EPS(1)⋅F⋅time_varying + F  for TIME < 1.0
```
Comment (rikardn) [inline src/pharmpy/modeling/error.py:626]: One space too much for the output

**Review by rikardn — APPROVED** [review]
Thanks! This is looking good now.

**Pull request merged**